### ISSUE TITLE
Problems setting up PHP_CodeSniffer in Emacs

### ISSUE THREAD

**Issue by charliecodesonline** [flycheck/flycheck#2046]
This template is for **user support**.  If you are reporting an actual Flycheck bug, please use the relevant issue template instead.

## Checklist
- [x] I have searched existing issues and StackOverflow for solutions.
- [x] I have read the [Flycheck manual][].
- [x] I have read the [Troubleshooting Guide][].
- [x] I have checked my issue has not been reported yet.

## Problem description
I have configured a custom ruleset and it is not getting used.

## Steps to reproduce
1. Add configuration to init.el
```clisp
(use-package flycheck
  :ensure t
  :diminish flycheck-mode
  :custom
  (flycheck-php-executable "/usr/bin/php")
  (flycheck-phpcs-executable "/usr/local/bin/phpcs")
  (flycheck-phpcs-standard "/usr/local/bin/.phpcs.xml")
  (flycheck-phpmd-executable "/usr/local/bin/phpmd")
  :init (global-flycheck-mode))
(add-hook 'after-init-hook #'global-flycheck-mode)
```
3. Create /usr/local/bin/.phpcs.xml
```xml
<?xml version="1.0"?>
<ruleset name="SERStandard">
    <description>My custom coding standard.</description>
    <rule ref="PEAR">
        <exclude name="PEAR.NamingConventions.ValidFunctionName.PrivateNoUndescore" />
    </rule>
</ruleset>
```
PEAR.NamingConventions.ValidFunctionName error still occurs in mini buffer window
- [x] I have read https://emacs.stackexchange.com/questions/28429/how-do-i-troubleshoot-emacs-problems

## Expected behavior
The rule should no longer be enforced so this error should not be displayed in the minibuffer

## Screenshots

![image](https://github.com/flycheck/flycheck/assets/151664451/f987274e-e139-44ef-83bf-47a8dc539bf1)

## System configuration
```
Syntax checkers for buffer AlphaCharacterState.php in php-mode:

First checker to run:

  php
    - may enable:    yes
    - executable:    Found at /usr/bin/php
    - next checkers: php-phpmd, php-phpcs

Checkers that may run as part of the first checker's chain:

  php-phpmd
    - may enable:    yes
    - executable:    Found at /usr/local/bin/phpmd
    - next checkers: php-phpcs

  php-phpcs
    - may enable: yes
    - may run:    t
    - executable: Found at /usr/local/bin/phpcs

Flycheck Mode is enabled.  Use C-u C-c ! x to enable disabled
checkers.

--------------------

Flycheck version: 33snapshot (package: 20230813.620)
Emacs version:    27.1
System:           x86_64-pc-linux-gnu
Window system:    x

```

Emacs configuration:
- [x] Plain Emacs / Custom configuration
- [ ] Spacemacs
- [ ] Doom Emacs
- [ ] Other shared configuration

## Additional notes

[Flycheck manual]:  https://www.flycheck.org/en/latest/user/troubleshooting.html
[Troubleshooting guide]:  https://www.flycheck.org/en/latest/user/troubleshooting.html

**Comment by bbatsov** [comment]
@flycheck/php Can someone take a look?
